frontend/device/kobo/device.lua

```diff
@@ -496,19 +496,15 @@ function Kobo:initNetworkManager(NetworkMgr)
     -- NOTE: Cheap-ass way of checking if Wi-Fi seems to be enabled...
     --       Since the crux of the issues lies in race-y module unloading, this is perfectly fine for our usage.
     function NetworkMgr:isWifiOn()
-        local fd = io.open("/proc/modules", "r")
-        if fd then
-            local lsmod = fd:read("*all")
-            fd:close()
-            -- lsmod is usually empty, unless Wi-Fi or USB is enabled
-            -- We could alternatively check if lfs.attributes("/proc/sys/net/ipv4/conf/" .. os.getenv("INTERFACE"), "mode") == "directory"
-            -- c.f., also what Cervantes does via /sys/class/net/eth0/carrier to check if the interface is up.
-            -- That said, since we only care about whether *modules* are loaded, this does the job nicely.
-            if lsmod:len() > 0 then
-                local module = os.getenv("WIFI_MODULE") or "sdio_wifi_pwr"
-                if lsmod:find(module) then
-                    return true
-                end
+        local needle = os.getenv("WIFI_MODULE") or "sdio_wifi_pwr"
+        local nlen = #needle
+        -- /proc/modules is usually empty, unless Wi-Fi or USB is enabled
+        -- We could alternatively check if lfs.attributes("/proc/sys/net/ipv4/conf/" .. os.getenv("INTERFACE"), "mode") == "directory"
+        -- c.f., also what Cervantes does via /sys/class/net/eth0/carrier to check if the interface is up.
+        -- That said, since we only care about whether *modules* are loaded, this does the job nicely.
+        for haystack in io.lines("/proc/modules") do
+            if haystack:sub(1, nlen) == needle then
```

Comment:
(Unless you're teaching programming, I'd call these needle and haystack what they are: _the widi module_ and _any module_.)
(And your haystack is not a stack, it's just a _brin_ taken from the stack as you're iterating each one from the stack :)

Reply:
c.f., answer below, there may be far more noise on each line than just a module noise ;).

The choice of words is *exactly* chosen to drive the point home that this is actually an `strncmp` in disguise ;).

Reply:
I'm inclined to agree it mostly just looks like programmerese but oh well. :-P